cmd/argocd/commands/admin/appset.go

```diff
@@ -0,0 +1,157 @@
+package admin
+
+import (
+	"io"
+	"os"
+
+	"github.com/argoproj/argo-cd/v2/applicationset/controllers/template"
+	"github.com/argoproj/argo-cd/v2/applicationset/generators"
+	"github.com/argoproj/argo-cd/v2/applicationset/services"
+	"github.com/argoproj/argo-cd/v2/applicationset/utils"
+	"github.com/argoproj/argo-cd/v2/cmd/argocd/commands/headless"
+	"github.com/argoproj/argo-cd/v2/cmd/argocd/commands/initialize"
+	"github.com/argoproj/argo-cd/v2/common"
+	"github.com/argoproj/argo-cd/v2/pkg/apiclient"
+	"github.com/argoproj/argo-cd/v2/pkg/apis/application/v1alpha1"
+	"github.com/argoproj/argo-cd/v2/util/cli"
+	"github.com/argoproj/argo-cd/v2/util/db"
+	"github.com/argoproj/argo-cd/v2/util/github_app"
+	ioutil "github.com/argoproj/argo-cd/v2/util/io"
+	"github.com/argoproj/argo-cd/v2/util/settings"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+)
+
+func NewAppsetCommand(opts *apiclient.ClientOptions) *cobra.Command {
```

Comment:
I feel like this does not make a lot of sense under the admin commands. Maybe `argocd appset template ./appset.yaml -o yaml` would make more sense. Take a look at https://github.com/argoproj/argo-cd/pull/16781, that has prior work to implement this feature.

Reply:
Agree - moved command to `argocd appset generate`. I've choosen to name it `generate` , however don't have strong opinion . Please let me know what you think